src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.lib.exception.DataProcessingException;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
```

Comment:
```suggestion
            transaction.commit();
            return movie;
        } catch (Exception e) {
```
better to 'return' in try block, if this try block is the last piece of logic in your code

Reply:
done